### DISCUSSION TITLE
rqlite is Intolerably slow in my macbook

### DISCUSSION THREAD

**Discussion by koftnt** [rqlite/rqlite#1365] [Q&A]
hi all.
I am a green hand for rqlite，I installed it yesterday following the use guide.
I aim to  replace local sqlite db by using rqlite for distribution。But in my macbook local test . Rqlite is very slow . I think this is abnormal.
I use sqlalchemy-rqlite to create engine 
![image](https://github.com/rqlite/rqlite/assets/16820678/579c6d0c-b2ea-4143-830c-a32de8fa98ce)
there is only one local node in my mac
![image](https://github.com/rqlite/rqlite/assets/16820678/be73cac7-d78e-4310-a7d2-b3fd74f4adeb)
![image](https://github.com/rqlite/rqlite/assets/16820678/833efc3a-73c4-44e9-9f6d-9dd8fb7835b5)

Here are  some same sql query time cost using rqlite in different time, the time unit in following picture is second.
![image](https://github.com/rqlite/rqlite/assets/16820678/4323562b-9323-44a2-81e3-c2e0b035c69c)
![image](https://github.com/rqlite/rqlite/assets/16820678/13181448-012e-4e5d-abef-a4fdfcbdd97a)
![image](https://github.com/rqlite/rqlite/assets/16820678/57bb2298-a0fd-4385-8ce6-6b25a6dae022)
![image](https://github.com/rqlite/rqlite/assets/16820678/78ee8b75-fae5-4e49-8b95-d5aa1d156ec3)
![image](https://github.com/rqlite/rqlite/assets/16820678/ebe4d8f0-2bfb-43ad-845b-40ee5334805b)

But the similar sql query in local sqlite db have a low time consuming。usually less than 2s
<img width="879" alt="image" src="https://github.com/rqlite/rqlite/assets/16820678/1bf7bef4-1bad-4b93-8c0a-171aabaacc5e">

**Comment by otoolep** [comment]
This is doing to be difficult to debug unless you tell me more information. I need to know the SQL you are executing, if it's the same database when using SQLite vs. rqlite and so on.

Take your python code out of the loop, and perform your queries directly using the HTTP API: https://rqlite.io/docs/api/api/ That will help use understand if it's rqlite or something else.

By the way, screenshots are an awkward way to share information. Simple copy 'n' paste is better.

**Comment by otoolep** [comment]
Any updates? If there are none, I may close this issue.

**Comment by otoolep** [comment]
Closing, no updates.
